Parsers/ASimWebSession/Parsers/vimWebSessionSonicWallFirewall.yaml

```diff
@@ -0,0 +1,450 @@
+Parser:
+  Title: Web Session ASIM filtering parser for SonicWall firewalls
+  Version: '0.1.0'
+  LastUpdated: Feb 28, 2024
+Product:
+  Name: SonicWall
+Normalization:
+  Schema: WebSession
+  Version: '0.2.6'
+References:
+  - Title: ASIM Web Session Schema
+    Link: https://aka.ms/ASimWebSessionDoc
+  - Title: ASIM
+    Link: https:/aka.ms/AboutASIM
+  - Title: SonicWall SonicOS/X 7.0.1 Log Events Reference Guide [PDF]
+    Link: https://www.sonicwall.com/techdocs/pdf/SonicOS-X_7.0.1_LogEvents_ReferenceGuide.pdf
+Description: |
+  This ASIM parser supports filtering and normalizing SonicWall SonicOS ArcSight-formatted syslog ingested by Microsoft Sentinel to the ASIM WebSession normalized schema.
+ParserName: vimWebSessionSonicWallFirewall
+EquivalentBuiltInParser: _Im_WebSession_SonicWallFirewall
+ParserParams:
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: srcipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: ipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: url_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: httpuseragent_has_any
+    Type: dynamic 
+    Default: dynamic([])
+  - Name: eventresultdetails_in
+    Type: dynamic
+    Default: dynamic([])
+  - Name: eventresult
+    Type: string
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let parser=(
+    starttime:datetime=datetime(null),
+    endtime:datetime=datetime(null),
+    srcipaddr_has_any_prefix:dynamic=dynamic([]),
+    ipaddr_has_any_prefix:dynamic=dynamic([]), 
+    url_has_any:dynamic=dynamic([]),
+    httpuseragent_has_any:dynamic=dynamic([]),
+    eventresultdetails_in:dynamic=dynamic([]),
+    eventresult:string='*',
+    disabled:bool=false
+    )
+    {
+      let src_or_any=set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix);
+      let Actions=datatable(fw_action:string, DvcAction:string, EventSeverity:string)
+        [ "\"forward\"", "Allow", "Informational"
+        , "\"mgmt\"", "Other", "Informational"
+        , "\"NA\"", "Other", "Informational"
+        , "\"drop\"", "Drop", "Low"
+      ];
+      CommonSecurityLog
+      | where not(disabled)
+      | where DeviceVendor == "SonicWall"
+      | where DeviceEventClassID in (14, 97)
+      | where (isnull(starttime) or TimeGenerated >= starttime) and (isnull(endtime) or TimeGenerated <= endtime)
+      | where (isnotempty(SourceIP) or isnotempty(DestinationIP))

```

Comment:
can we add pre-filter for url? like below?
and (array_length(url_has_any) == 0 or AdditionalExtensions  has_any (url_has_any))?

Reply:
Thanks for this tip. I thought I'd have to parse it first, but has_any worked as a general search of the field, which seems to be sufficient.

Reply:
at the end we will have a pre and post filter for Url